### PR TITLE
Update pytorch version

### PR DESCRIPTION
**Pull request by JamesLYC88**
See the [document](https://hackmd.io/h5nPzjs3RTKdDPC4AEXb_g) for more details.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,10 +2,10 @@ nltk
 pandas
 PyYAML
 scikit-learn
-torch==1.8.0
-torchmetrics==0.6.0
-torchtext==0.9.0
-pytorch-lightning==1.3.5
+torch
```
Comment (Eleven1Liu) [inline requirements.txt:5]: Should we leave the version numbers here? Sometimes the installation failed due to torchtext update, it changes a lot between versions (Ex. from 0.8.0 to 0.9.0).

File: libmultilabel/nn/data_utils.py [hunk 2]
```diff
@@ -221,46 +222,31 @@ def load_or_build_text_dict(
...
+        tuple[torchtext.vocab.Vocab, torch.Tensor]: A vocab object which maps tokens to indices and the pre-trained word vectors of shape (vocab_size, embed_dim).
     """
     if vocab_file:
         logging.info(f'Load vocab from {vocab_file}')
         with open(vocab_file, 'r') as fp:
-            vocab_list = [PAD] + [vocab.strip() for vocab in fp.readlines()]
-        vocabs = Vocab(collections.Counter(vocab_list), specials=[UNK],
-                       min_freq=1, specials_first=False)  # specials_first=False to keep PAD index 0
+            vocab_list = [[PAD] + [vocab.strip() for vocab in fp.readlines()]]
+        # special_first=False to keep PAD index 0
```
Comment (Eleven1Liu) [inline libmultilabel/nn/data_utils.py:231]: I think it's for aligning `padding_idx` [here](https://github.com/ASUS-AICS/LibMultiLabel/blob/ebd37b3c2ce05662e83d3c86a418bc0594904192/libmultilabel/nn/networks/modules.py#L20).

File: requirements.txt (hunk 1 above)
Comment (JamesLYC88) [reply requirements.txt]: Sure, thanks for pointing out. Users can also know the actual version here.

File: libmultilabel/nn/data_utils.py (hunk 2 above)
Comment (JamesLYC88) [reply libmultilabel/nn/data_utils.py:231]: Got it, thanks!

File: libmultilabel/nn/nn_utils.py [hunk 3]
```diff
@@ -170,7 +171,7 @@ def set_seed(seed):
 
     if seed is not None:
         if seed >= 0:
-            seed_everything(seed=seed)
+            seed_everything(seed=seed, workers=True)
             torch.use_deterministic_algorithms(True)
             torch.backends.cudnn.benchmark = False
```
Comment (sian-chen) [inline libmultilabel/nn/nn_utils.py:176]: Check if we still need these two lines.

File: libmultilabel/nn/data_utils.py [hunk 4]
```diff
@@ -221,46 +222,36 @@ def load_or_build_text_dict(
...
+        tuple[torchtext.vocab.Vocab, torch.Tensor]: A vocab object which maps tokens to indices and the pre-trained word vectors of shape (vocab_size, embed_dim).
     """
     if vocab_file:
         logging.info(f'Load vocab from {vocab_file}')
         with open(vocab_file, 'r') as fp:
-            vocab_list = [PAD] + [vocab.strip() for vocab in fp.readlines()]
-        vocabs = Vocab(collections.Counter(vocab_list), specials=[UNK],
-                       min_freq=1, specials_first=False)  # specials_first=False to keep PAD index 0
+            vocab_list = [[vocab.strip() for vocab in fp.readlines()]]
+        # special_first=False to keep PAD index 0
```
Comment (cjlin1) [inline libmultilabel/nn/data_utils.py:231]: I think Eleven meant this comment needs to be changed as we don't have "special_first" any more

File: libmultilabel/nn/nn_utils.py (hunk 3 above)
Comment (JamesLYC88) [reply libmultilabel/nn/nn_utils.py]: These lines are not necessary. Updated, thanks!

File: libmultilabel/nn/data_utils.py (hunk 4 above)
Comment (JamesLYC88) [reply libmultilabel/nn/data_utils.py]: Thanks. The comment has now been updated to a notification that we should keep PAD index to 0.

File: libmultilabel/nn/data_utils.py [hunk 5]
```diff
@@ -327,6 +327,13 @@ def get_embedding_weights_from_file(word_dict, embed_file, silent=False, cache=N
 
     embedding_weights = torch.zeros(len(word_dict), embed_size)
 
+    if load_embedding_from_file:
+        # Add UNK embedding
+        # AttentionXML: np.random.uniform(-1.0, 1.0, embed_size)
+        # CAML: np.random.randn(embed_size)
+        unk_vector = torch.randn(embed_size)
+        embedding_weights[word_dict[UNK]] = unk_vector
```
Comment (JamesLYC88) [inline libmultilabel/nn/data_utils.py:335]: Sorry for the confusion.
For loading pre-trained vector like glove, we do not initialize the UNK vector ourselves but simply use the UNK vector provided in glove vector_dict.
On the other hand, for data like MIMIC, we follow CAML to randomly initialize a UNK vector.

File: torch_trainer.py [hunk 6]
```diff
@@ -177,6 +176,9 @@ def train(self):
 
         if 'val' not in self.datasets:
             logging.info('No validation dataset is provided. Train without vaildation.')
+            from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+            self.trainer.callbacks = \
+                [callback for callback in self.trainer.callbacks if not isinstance(callback, EarlyStopping)]
```
Comment (sian-chen) [inline torch_trainer.py:181]: Why apply early stopping when no validation is provided?

Comment (JamesLYC88) [reply torch_trainer.py:181]: This logic is to keep the callbacks that are **not** early stopping if no validation provided.

**Review by sian-chen — APPROVED** [review]

**Pull request merged**